Document the changed default AMQP port value

chris-janidlo opened the pull request:
# Description

These changes were made in the AWS console via updates to secrets manager values. See `compute-<env>-mq-admin` secrets for details.

[[sc-33254]](https://app.shortcut.com/globus/story/33254/change-default-port-for-the-amqps-port-provided-by-the-webservice-to-the-executor)

Note that prod hasn't been updated, per the comment on the story. I'll wait to update that until release day.

## Type of change

- Documentation update

rjmello commented on compute_endpoint/globus_compute_endpoint/endpoint/config/config.py:
```diff
@@ -129,7 +129,7 @@ class Config(RepresentationMixin):
         supported by the Compute web services. If None, the port is assigned by the
-        services (typically 5671).
+        services (which default to 443).
         Default: None
```
Although redundant, I think we should set the default value of the attribute to 443 for clarity. This will save the user from having to understand the particular interaction between the SDK and web service.

LeiGlobus commented on compute_endpoint/globus_compute_endpoint/endpoint/config/config.py:
```diff
@@ -129,7 +129,7 @@ class Config(RepresentationMixin):
         Port to use for AMQP connections. Note that only 5671, 5672, and 443 are
         supported by the Compute web services. If None, the port is assigned by the
-        services (typically 5671).
+        services (which default to 443).
```
It would help future devs (or us) if there's a hint where this default is set. ie. 

"which default to 443 from the URL passed back, constructed from FUNCX_AMQP_URL in the services"

Update - Just saw Reid's comment, explicit assigning to 443 here does help clarity, but that would change code logic, and make setting :443 in the service URL redundant.  will discuss in # dev

chris-janidlo replied:
Decided against this since the docstrings are nominally user facing

LeiGlobus commented on changelog.d/20240604_170320_chris_document_443_default_switch.rst:
```diff
@@ -0,0 +1,7 @@
+
+- The Compute hosted services now tell the SDK to connect to AMQP over port 443 by
+  default. This means that ``Executor(endpoint_id=..., amqp_port=443)`` is now redundant.
+  To go back to the old port of 5671, set ``Executor.amqp_port`` to 5671 via the
```
I think "To go back to the old" may be confusing without context, as not everyone is upgrading, so 'back to' may not always make senes.

How about "To use the standard AMQP port, set ...."

chris-janidlo replied:
Rewrote the changelog to be more friendly to end users

LeiGlobus reviewed (dismissed):
Letting Chris make the decision, random() or otherwise...

chris-janidlo commented:
Decided to not make `443` the default in code because it's a little awkward to explain in doc strings. Also updated the commit message and removed `amqp_port: 443` from the default config.

LeiGlobus reviewed (approved):
Less code is good code?
